fpmsyncd/routesync.cpp

```diff
@@ -756,88 +779,145 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
             return;
     }
 
-    struct nl_list_head *nhs = rtnl_route_get_nexthops(route_obj);
-    if (!nhs)
-    {
-        SWSS_LOG_INFO("Nexthop list is empty for %s", destipprefix);
-        return;
-    }
-
-    /* Get nexthop lists */
+    vector<FieldValueTuple> fvVector;
     string gw_list;
     string intf_list;
     string mpls_list;
-    getNextHopList(route_obj, gw_list, mpls_list, intf_list);
-    string weights = getNextHopWt(route_obj);
 
-    vector<string> alsv = tokenize(intf_list, NHG_DELIMITER);
-    for (auto alias : alsv)
+    string nhg_id_key;
+    uint32_t nhg_id = rtnl_route_get_nh_id(route_obj);
+    if(nhg_id)
     {
-        /*
-         * An FRR behavior change from 7.2 to 7.5 makes FRR update default route to eth0 in interface
-         * up/down events. Skipping routes to eth0 or docker0 to avoid such behavior
-         */
-        if (alias == "eth0" || alias == "docker0")
+        const auto itg = m_nh_groups.find(nhg_id);
+        if(itg == m_nh_groups.end())
+        {
+            SWSS_LOG_DEBUG("NextHop group id %d not found. Dropping the route %s", nhg_id, destipprefix);
+            return;
+        }
+        NextHopGroup& nhg = itg->second;
+        if(nhg.group.size() == 0)
         {
-            SWSS_LOG_DEBUG("Skip routes to eth0 or docker0: %s %s %s",
-                    destipprefix, gw_list.c_str(), intf_list.c_str());
-            // If intf_list has only this interface, that means all of the next hops of this route 
-            // have been removed and the next hop on the eth0/docker0 has become the only next hop. 
-            // In this case since we do not want the route with next hop on eth0/docker0, we return. 
-            // But still we need to clear the route from the APPL_DB. Otherwise the APPL_DB and data 
-            // path will be left with stale route entry
-            if(alsv.size() == 1)
+            //Using route-table only for single next-hop
+            string nexthops, ifnames, weights;
+
+            getNextHopGroupFields(nhg, nexthops, ifnames, weights, rtnl_route_get_family(route_obj));
```

Comment:
Do you want to handle a case where the nhg is not based on the ID?
If there is a failure  in `getNextHopGroupFields()`, you would be pushing empty strings.

Reply:
We need FRR to provide NHG ID. Otherwise, the NHG key would be changed during path changes.

Reply:
As eddie says, FRR is needed to provide NHG IDs. There is no case where nhg is not based on identity,